ExampleApp/ExampleApp/Screens/ScanResults/ViewController/ScanResultsViewController.swift

```diff
@@ -0,0 +1,134 @@
+import RxBluetoothKit
+import RxCocoa
+import RxSwift
+import UIKit
+
+class ScanResultsViewController: UIViewController, CustomView {
+
+    typealias ViewClass = BaseView
+
+    typealias ScansResultDataSource = TableViewDataSource<ScannedPeripheral, ScanResultsViewModelItem>
+
+    private let dataSource: TableViewDataSource<ScannedPeripheral, ScanResultsViewModelItem>
+
+    private let viewModel: ScanResultsViewModelType
+
+    private let disposeBag: DisposeBag = DisposeBag()
+
+    init(with dataSource: ScansResultDataSource, viewModel: ScanResultsViewModelType) {
+        self.dataSource = dataSource
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func loadView() {
+        view = ViewClass()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        setNavigationBar()
+        bindViewModel()
+    }
+
+    private func setupTableView() {
+        customView.setTableView(dataSource: dataSource, delegate: self)
+        registerCells()
+        setDataSourceRefreshBlock()
```

Comment:
as before i would put implementation of this functions here rather than calling them :)